src/sprig-matrix.ts

```diff
@@ -0,0 +1,222 @@
+import * as GLM from "gl-matrix.js";
+
+interface Float32ArrayOfLength<N extends number> extends Float32Array {
+  length: N;
```

Comment:
oh cool! Does this type check e.g. assigning [2,2] vs [2,2,2] ?

Reply:
Unfortunately, it doesn't really. If we want a `Float32ArrayOfLength<3>`, typechecking will fail if we pass in `new Float32Array(2)` but it will also fail if we pass in `new Float32Array(3)` because typescript doesn't know anything about the length of either. So, we end up needing to cast. Not sure it actually ends up being better than using some fake nominal-typing trick like https://stackoverflow.com/a/61296495 but this does feel a bit better.